app/api/observations/observations.utils.js

```diff
@@ -24,12 +24,9 @@ exports.getOwnerScope = req => {
       attributes: [],
       where
     });
-  } else if (user.hasRole('volunteer')) {
-    where.user_id = user.id;
   } else if (!user.isSuperAdmin()) {
-    // A super administrator can do anything. Any other user should not reach
-    // this point.
-    throw new Error(`Cannot determine owner scope for user ${user.id} with unsupported role(s) ${user.roles.join(', ')}`);
+    // scope volunteers to see only their own observations
+    where.user_id = user.id;
```

Comment:
what if anonymous user id 14, Then any anonymous user will see other anonymous users observations .

Reply:
anonymous request do not have a user in the request. You can check the authenticate middleware
```
exports.authenticate = (options = {}) => {
  const authenticationRequired = get(options, 'required', true);
  return (req, res, next) => passport.authenticate("jwt", (err, user) => {
    if (err) {
      // An authentication error occurred.
      next(err);
    } else if (user) {
      // The user was successfully authenticated.
      req.user = user;
      next();
    } else if (!user && !authenticationRequired && !req.get('Authorization')) {
      // No user was authenticated but authentication is optional and the
      // Authorization header was not sent.
      next();
    } else {
      // Authentication failed (e.g. the token is invalid or has expired).
      next(authenticationError(req.__('auth.error.generalServerAuth')));
    }
  })(req, res, next);
};
```